test/app.js

```diff
@@ -251,7 +241,59 @@ describe('Smoke test', () => {
         should(app.isNetStateConnected()).be.true();
         should(app.isNetStateDisconnected()).be.false();
         should(app.networkState).be.equal('Connected');
+      });
+  }).timeout(5000);
+
+  it('network state upon network disconnection event', (done) => {
+    let cbCount = 0;
+    const networkCb = () => {
+      if (cbCount === 1) {
+        should(app.networkState).be.equal('Disconnected');
+        done();
+      }
+      cbCount++; // eslint-disable-line
+    };
+
+    const appConfig = new App({
+      id: 'net.maidsafe.test.javascript.id',
+      name: 'NodeJS Test',
+      vendor: 'MaidSafe.net Ltd'
+    }, networkCb, { log: false });
+    const app = autoref(appConfig);
+
+    app.auth.loginFromURI(h.authUris.registeredUri)
+      .then(() => {
+        should(app.networkState).be.equal('Connected');
         app.auth.simulateNetworkDisconnect();
       });
   }).timeout(5000);
+
+  it('network state upon network reconnect event', (done) => {
+    let cbCount = 0;
+    const networkCb = () => {
+      if (cbCount === 1) {
```

Comment:
A more cleaner way is achieved with a switch statement, or with an array/map which contains the results expected for each `cbCount`. Perhaps worthy verifying the case for `cbCount === 0` as well.